src/klr.net45/klr.net45.cpp

```diff
@@ -8,20 +8,25 @@
 
 IKatanaManagerPtr g_katanaManager;
 
-
 extern "C" __declspec(dllexport) HRESULT __stdcall CallApplicationMain(PCALL_APPLICATION_MAIN_DATA data)
 {
     HRESULT hr = S_OK;
-    
+
     IKatanaManagerPtr manager = new ComObject<KatanaManager, IKatanaManager>();
 
     g_katanaManager = manager;
-    _HR(manager->InitializeRuntime(data->applicationBase));
-    g_katanaManager = NULL;
-
-    _HR(manager->CallApplicationMain(data->argc, data->argv));
 
-    data->exitcode = hr;
+    hr = manager->InitializeRuntime(data->applicationBase);
+    if (SUCCEEDED(hr))
+    {
+        g_katanaManager = NULL;
+        data->exitcode = manager->CallApplicationMain(data->argc, data->argv);
+    }
+    else 
+    {
+        printf_s("Failed to initalize runtime (%x)", hr);
```

Comment:
After consult the code in `klr.core45`, I choose to print the message here instead of outer `klr` caller. For two reasons:

1. [`klr.core45.cpp`](https://github.com/aspnet/KRuntime/blob/dev/src/klr.core45/klr.core45.cpp) already prints all kind of errors.
2. There is advantage to print the error message here. Since it is close to the crash site the error can contain meaningful information. Returning `HRESULT` to caller doesn't necessary explain everything.